src/allmydata/storage/http_server.py

```diff
@@ -562,12 +566,21 @@ def _read_encoded(
             message = mmap.mmap(fd, 0, access=mmap.ACCESS_READ)
         else:
             message = request.content.read()
-        schema.validate_cbor(message)
+
+        # Pycddl will release the GIL when validating larger documents, so
+        # let's take advantage of multiple CPUs:
+        if size > 10_000:
+            await deferToThread(reactor, schema.validate_cbor(message))
```

Comment:
Can `HTTPServer` have a reactor attribute to use here instead of relying on the global reactor?

Reply:
Turns out `deferToThread` doesn't actually take a reactor argument. So...

Reply:
I guess given Twisted's thread pool is used for blocking operations that aren't typically CPU intensive, better to have a different thread pool with size matching the number of cores. And then we can use `deferToThreadPool` which does a `reactor` argument.